src/ipc/handler.c

```diff
@@ -1230,20 +1232,70 @@ static int ipc_glb_tplg_message(uint32_t header)
 	}
 }
 
-#if CONFIG_DEBUG
+static int fill_mem_usage_elems(int zone, int elem_number,
+				struct sof_ipc_dbg_mem_usage_elem *elems)
+{
+	struct mm_info info;
+	int ret;
+	int i;
+
+	for (i = 0; i < elem_number; ++i) {
+		ret = heap_info(zone, i, &info);
+		elems[i].zone = zone;
+		elems[i].id = i;
+		elems[i].used = ret < 0 ? UINT32_MAX : info.used;
+		elems[i].free = ret < 0 ? 0 : info.free;
+	}
+
+	return elem_number;
+}
+
+static int ipc_glb_test_mem_usage(uint32_t header)
+{
+	int elem_cnt = PLATFORM_HEAP_SYSTEM + PLATFORM_HEAP_SYSTEM_RUNTIME +
+		       PLATFORM_HEAP_RUNTIME + PLATFORM_HEAP_BUFFER;
```

Comment:
What does this sum even mean? I feel some comments are warranted.

Reply:
```
/* heap block memory map */
struct mm {
	/* system heap - used during init cannot be freed */
	struct mm_heap system[PLATFORM_HEAP_SYSTEM];
	/* system runtime heap - used for runtime system components */
	struct mm_heap system_runtime[PLATFORM_HEAP_SYSTEM_RUNTIME];
	/* general heap for components */
	struct mm_heap runtime[PLATFORM_HEAP_RUNTIME];
	/* general component buffer heap */
	struct mm_heap buffer[PLATFORM_HEAP_BUFFER];

	struct mm_info total;
	uint32_t heap_trace_updated;	/* updates that can be presented */
	spinlock_t lock;	/* all allocs and frees are atomic */
};
```